game-core/src/test/java/games/strategy/triplea/delegate/battle/steps/fire/aa/DefensiveAaFireTest.java

```diff
@@ -0,0 +1,66 @@
+package games.strategy.triplea.delegate.battle.steps.fire.aa;
+
+import static games.strategy.triplea.delegate.battle.FakeBattleState.givenBattleStateBuilder;
+import static games.strategy.triplea.delegate.battle.steps.BattleStepsTest.givenUnitWithTypeAa;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+
+import games.strategy.engine.data.Unit;
+import games.strategy.engine.delegate.IDelegateBridge;
+import games.strategy.triplea.delegate.ExecutionStack;
+import games.strategy.triplea.delegate.battle.BattleActions;
+import games.strategy.triplea.delegate.battle.BattleState;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class DefensiveAaFireTest {
+
+  @Mock ExecutionStack executionStack;
+  @Mock IDelegateBridge delegateBridge;
+  @Mock BattleActions battleActions;
+
+  @ParameterizedTest(name = "[{index}] {0} is {2}")
```

Comment:
Have a second look at the parameterization if you would. I think it's a clear win if we have ballpark 5 or more variations. With just two, particularly with boolean logic, it might be more straight forward just enumerating the different test cases with individual test blocks. I think in big part parameterized tests should not have `if` blocks, that indicates really that you have multiple test methods (essentially one is testing too many things at once), secondarily data that changes how a test method behaves I think is arguably a test code-smell. WDYT?

Reply:
I could get rid of the `if` statement by adding another argument that has the number of names to expect.  So I don't think `if` is that big of a code smell.  I didn't add an extra argument because it was easy to determine what the number of names is based on the expected value.

But having multiple `asserts` that are tangentially related (valid and names) is more of a code smell in doing to much in the test.  I'll split them up.

`AirAttackVsNonSubsStepTest`, `AirDefendVsNonSubsStepTest`, and `SubmergeSubsVsOnlyAirStepTest` are all like this.  Should I change them as well?